### ISSUE TITLE
PoolFormer Patch Embedding is faster than EfficientFormer Patch Embedding on CPU and GPU

### ISSUE THREAD

**Issue by iumyx2612** [snap-research/EfficientFormer#30]
I re-implement PoolFormer first Patch Embedding (PE) as follow:
```
stem_big = nn.Sequential(
    nn.Conv2d(in_channels=3,
              out_channels=64,
              kernel_size=7,
              padding=2,
              stride=4)
)
```
And I also re-implement EffcientFormer PE as follow:
```
stem_smol = nn.Sequential(
    nn.Conv2d(3, 64 // 2, kernel_size=3, stride=2, padding=1),
    nn.BatchNorm2d(64 // 2),
    nn.ReLU(),
    nn.Conv2d(64 // 2, 64, kernel_size=3, stride=2, padding=1),
    nn.BatchNorm2d(64),
    nn.ReLU()
)
```
This is my function to measure FLOPS, Params and inference time: 
```
def get_model_info(model: nn.Module, tsize) -> str:
    from thop import profile
    from copy import deepcopy
    import time
    import torch

    stride = 224
    img = torch.zeros((tsize[0], tsize[1], stride, stride), device=next(model.parameters()).device)
    start = time.time()
    flops, params = profile(deepcopy(model), inputs=(img,), verbose=False)
    end = time.time()
    delta = end - start
    params /= 1e6
    flops /= 1e9
    flops *= tsize[2] * tsize[3] / stride / stride * 2  # Gflops
    info = "Params: {:.6f}M, Gflops: {:.2f}, time: {:.6f}".format(params, flops, delta)
    return info
```

I tested both module on CPU and GPU:
- My CPU: Intel(R) Core(TM) i5-7400 CPU @ 3.00GHz
- My GPU: NVIDIA RTX 3090

CPU inference time:
- PoolFormer PE: Params: 0.009472M, Gflops: 0.06, time: 0.003800
- EfficientFormer PE: Params: 0.019584M, Gflops: 0.14, time: 0.004068

GPU inference time:
- PoolFormer PE: Params: 0.0095M, Gflops: 0.059408, time: 0.000948
- EfficientFormer PE: Params: 0.0196M, Gflops: 0.140894, time: 0.002671

This is significantly slower than PoolFormer PE? Is there anything wrong in my implementation?

**Comment by alanspike** [comment]
Hi @iumyx2612, thanks for your interests. Could you try the official implementation of poolformer from their [repo](https://github.com/sail-sg/poolformer)?

**Comment by iumyx2612** [comment]
> Hi @iumyx2612, thanks for your interests. Could you try the official implementation of poolformer from their [repo](https://github.com/sail-sg/poolformer)?

My implementation of `stem_big` is almost the same as [PoolFormer implementation](https://github.com/sail-sg/poolformer/blob/main/models/poolformer.py#L64), which consists of a `nn.Conv2D` with kernel size, stride and padding is 7, 4 and 2 respectively: https://github.com/sail-sg/poolformer/blob/aa44f875bbcd975f455a597f4e46efe7fd94be3e/models/poolformer.py#L259

**Comment by alanspike** [comment]
Could you please try them on mobile devices if possible? Large kernel size is not mobile-friendly.

**Comment by iumyx2612** [comment]
> Could you please try them on mobile devices if possible? Large kernel size is not mobile-friendly.

oh my concern was if your stem implementation is faster than normal stem on **CPU and GPU**. I had zero programming experience with mobile device and mobile deployment. So this means your two 3x3 Conv stem is slower than a large kernel stem on CPU and GPU devices right?

**Comment by alanspike** [comment]
Hi @iumyx2612 , for your implementation of the comparisons, the two 3x3 Conv ops is slower than the 7x7 Conv on server CPU/GPU with Pytorch.

**Comment by iumyx2612** [comment]
> Hi @iumyx2612 , for your implementation of the comparisons, the two 3x3 Conv ops is slower than the 7x7 Conv on server CPU/GPU with Pytorch.
Thank you for your clarification!
